docs/pages/api-reference/liveblocks-react-comments.mdx

```diff
@@ -0,0 +1,852 @@
+
+const client = createClient({
+  /* client options */
+});
+
+// Presence represents the properties that will exist on every User in the Room
+// and that will automatically be kept in sync. Accessible through the
+// `user.presence` property. Must be JSON-serializable.
+type Presence = {
+  // cursor: { x: number, y: number } | null,
+  // ...
+};
+
+// Optionally, Storage represents the shared document that persists in the
+// Room, even after all Users leave. Fields under Storage typically are
+// LiveList, LiveMap, LiveObject instances, for which updates are
+// automatically persisted and synced to all connected clients.
+type Storage = {
+  // animals: LiveList<string>,
+  // ...
+};
+
+// Optionally, UserMeta represents static/readonly metadata on each User, as
+// provided by your own custom auth backend (if used). Useful for data that
+// will not change during a session, like a User's name or avatar.
+type UserMeta = {
+  id?: string; // Accessible through `user.id`
+  info?: Json; // Accessible through `user.info`
+};
+
+// Optionally, the type of custom events broadcasted and listened for in this
+// room. Must be JSON-serializable.
+type RoomEvent = {};
+
+// Optionally, ThreadMetadata represents metadata on each thread.
+// Must only contain scalar values.
+type ThreadMetadata = {
+  resolved?: boolean;
+};
+
+export const {
+  RoomProvider,
+  useThreads,
+  /* ...all the other hooks you’re using... */
+} = createRoomContext<Presence, Storage, UserMeta, RoomEvent, ThreadMetadata>(
+  client
+);
+```
+
+#### Resolving users [#createRoomContext.resolveUser]
+
+Anything that is linked to a user (comments, mentions, etc.) will reference
+their ID. To display user information based on IDs, you can provide a resolver
+function to the `resolveUser` option in
+[`createRoomContext`](#createRoomContext).
+
+This resolver function will receive a user ID and should return a user object
+asynchronously.
+
+```tsx file="liveblocks.config.ts"
+// ...
+
+export const {
+  /* ... */
+} = createRoomContext(client, {
+  resolveUser: (userId) => __getUserInfoFromDB__(userId),
+});
+```
+
+Once that’s done, components will show user names and avatars accordingly.
+
+#### Resolving mention suggestions [#createRoomContext.resolveMentionSuggestions]
+
+Liveblocks only knows users’ IDs, so to enable creating mentions in composer
+editors, you can provide a resolver function to the `resolveMentionSuggestions`
+option in [`createRoomContext`](#createRoomContext).
+
+This resolver function will receive the mention currently being written (e.g.
+when writing “@jane”, `text` will be `jane`) and should return a list of user
+IDs matching that text. This function will be called every time the text changes
+but with some debouncing.
+
+```tsx file="liveblocks.config.ts"
+// ...
+
+export const {
+  /* ... */
+} = createRoomContext(client, {
+  resolveMentionSuggestions: (text) => __getSuggestedUserIds__(text),
```

Comment:
Similar to above, can we show what we're returning (then perhaps have a collapsed more realistic example below).

Reply:
I've tweaked the snippet.